### PR TITLE
C2PA-37: Do not perform XMP writes for fonts.

### PR DESCRIPTION
**Pull request by brogdonm**
## Changes in this pull request
_Give a narrative description of what has been changed._

When adding support for font types, we had to temporarily disable the XMP writing, since it is not valid for a Font file. This change allows for the `xmp_write` feature to be used with the `otf` feature. In essence allowing for a single compile of the SDK to allow writing manifests to a font file and an image file.

The `c2patool` has a draft PR referencing this branch for testing: https://github.com/Monotype/c2patool/pull/1.

## Checklist
- [X] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/jumbf_io.rs [hunk 1]
```diff
@@ -74,6 +74,19 @@ static BMFF_TYPES: [&str; 12] = [
     "video/mp4",
 ];
 
+#[cfg(all(feature = "otf", feature = "file_io"))]
+static FONT_TYPES: [&str; 4] = [
```
Comment (chriskushmonotype) [inline sdk/src/jumbf_io.rs]: I notice that the `"otf"` feature is referenced in `sdk/src/jumbf_io.rs`, like so:

```
   // yadda yadda yadda
   match asset_type.as_ref() {
        // yadda yadda yadda
        "png" | "image/png" => Some(Box::new(PngIO {})),
        #[cfg(feature = "otf")]
        "otf" | "application/font-sfnt" | "ttf" | "font/ttf" => Some(Box::new(OtfIO {})),
        "avif" | "heif" | "heic" | "mp4" | "m4a" | "application/mp4" | "audio/mp4"
        // yadda yadda yadda
```

This arrangement seems brittle; is there some way that the line matching on the fonty MIME types could rely directly on the definition of `FONT_TYPES`?

File: sdk/src/jumbf_io.rs [hunk 2]
```diff
@@ -74,6 +74,19 @@ static BMFF_TYPES: [&str; 12] = [
...
+    "otf",
+    "ttf",
+    "application/font-sfnt",
+    "font/ttf",
```
Comment (chriskushmonotype) [inline sdk/src/jumbf_io.rs:88]: It would be scope creep to grow the list of MIME types that indicate a font, but [cursory research](https://stackoverflow.com/questions/2871655/proper-mime-type-for-otf-fonts/20723357#20723357) suggests this should probably be expanded (even leaving aside WOFF/2 support).

**Review by chriskushmonotype — COMMENTED** [review]
I tested manually & verified the change (my new build of `c2patool` can read & write manifests to both image & font files, whereas the basic `monotype/fontSupport` branch.

_(And after consulting with the author, they found a code path which now produces a misleading error message that it would be nice to fix -- will review those changes when they arrive.)_

File: sdk/src/jumbf_io.rs (hunk 2 above)
Comment (brogdonm) [reply sdk/src/jumbf_io.rs:88]: OK, I added the additional ones. I did not bother with WOFF at this point.

File: sdk/src/jumbf_io.rs (hunk 1 above)
Comment (brogdonm) [reply sdk/src/jumbf_io.rs]: I threw something in that accomplishes what you were getting at.

**Comment by brogdonm** [comment]
@chriskushmonotype the current code is a quick attempt to see what it would take to abstract the embedding of the URI to the external manifest and served more of a playground purpose. Truely abstracting this would require quite amount of refactoring and moving things around. For that reason, I have created an internal PBI for looking at it from that perspective. 

I do not plan to merge this, but I figured I would leave it open for your once review to facilitate an understanding of the requirement.

File: sdk/src/jumbf_io.rs (hunk 1 above)
Comment (chriskushmonotype) [reply sdk/src/jumbf_io.rs]: Excellent!

**Comment by brogdonm** [comment]
@chriskushmonotype I'm going to close this PR and just put in a very basic hack/fix for allowing for `xmp_write` feature and using fonts and the included extra mime-types and file extensions as you pointed out. I will then create another branch capturing my thoughts and findings for future refinement/planning.
